tests/tkupchyn/homework_08/test_warior.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+from tkupchyn.homework_08.warior import (Warrior, Knight, fight)
+
+
+@pytest.fixture()
+def duel_pairs():
+    return [(Warrior(), Warrior()),
+            (Warrior(), Knight()),
+            (Warrior(), Warrior()),
+            (Knight(), Knight())]
+
+
+def test_fight(duel_pairs):
+    expected_result = [True, False, True, True]
+    for i in range(len(duel_pairs)):
+        assert fight(duel_pairs[i][0], duel_pairs[i][1]) == expected_result[i]
+
+
+def test_is_alive(duel_pairs):
+    expected_result = [{'before_fight': True, 'after_fight': True},
+                       {'before_fight': True, 'after_fight': False},
+                       {'before_fight': True, 'after_fight': True},
+                       {'before_fight': True, 'after_fight': True}]
+    for i in range(len(duel_pairs)):
+        assert duel_pairs[i][0].is_alive == expected_result[i]['before_fight']
+        fight(duel_pairs[i][0], duel_pairs[i][1])
+        assert duel_pairs[i][0].is_alive == expected_result[i]['after_fight']
```

Comment:
I can suggest to rearange test like this:

```
def test_is_alive():
    assert Warior().is_alive

def test_duels(duel_pairs):
    assert fight(Warior(), Warior()) is True
    assert fight(Warior(), Knight()) is False
    assert fight(Knight(), Knight()) is True


```

Reply:
Thanks, it's more simple and readble.